packages/blockchain/src/index.ts

```diff
@@ -770,6 +772,10 @@ export default class Blockchain implements BlockchainInterface {
           lastBlock = block
           await onBlock(block, reorg)
           blockNumber.iaddn(1)
+          blocksRanCounter++
+          if (maxBlocks && blocksRanCounter == maxBlocks) {
```

Comment:
Two things here:

1. I guess this would be the only place in our whole codebase where the double equals would be used 😛, can we update to the standard triple equals?

2. The break condition is unnecessarily late in the loop (correct me if I am wrong), this gets most obvious when ran on an edge case (`maxBlocks` === 0) where code is executed which shouldn't be executed which could lead to side effects. This should be placed at the top (eventually just directly integrate into the while condition, something like `while (maxBlocks === undefined || maxBlocks && blocksRanCounter == maxBlocks)`)? Can also be done separately as well though.

Reply:
Should be fixed in 833264f 😄

Reply:
Have re-fixed in 5497bc0